core/src/world/npc/view.rs

```diff
@@ -97,7 +97,13 @@ impl<'a> fmt::Display for DetailsView<'a> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         let Self { npc, relations } = self;
 
-        writeln!(f, "<div class=\"thing-box npc\">\n")?;
+        writeln!(
+            f,
+            "<div class=\"thing-box npc\"{}>\n",
+            npc.uuid
+                .clone()
+                .map_or("".to_string(), |v| format!(" data-uuid=\"{}\"", v))
```

Comment:
I thiiink this is doable with [`format_args!()`](https://doc.rust-lang.org/std/macro.format_args.html), thereby avoiding a redundant allocation.

Reply:
There may be a more elegant way to express this but this is what I've come up with to avoid the allocation.

```rs
let uuid = npc.uuid.as_ref();

writeln!(
    f,
    "<div class=\"thing-box npc\"{}>\n",
    (uuid.is_some())
        .then_some(format_args!(" data-uuid=\"{}\"", uuid.unwrap()))
        .unwrap_or(format_args!(""))
)?;
```

There is an ownership issue with just dropping in `format_args!()` with the original solution, this manages to get around all of that. Inspired by [this thread](https://internals.rust-lang.org/t/format-args-with-long-lifetimes/19494/6).

Reply:
In fact, the type will probably change from `Option<Uuid>` to `Uuid` with the UUID-on-create change, rendering all of this moot.

Reply:
This is true!